### PR TITLE
Remove expression templates

### PR DESCRIPTION
**Pull request by jamiebramwell**
The expression template implementation was causing issues when interacting with the dual number class due to templated operator overloads. As a simpler implementation is possible and we would like consistent finite element space definitions, this removes the current expression template capability. It was mostly used for testing purposes.

### CODE REVIEW

File: src/serac/numerics/functional/element_restriction.cpp [hunk 1]
```diff
@@ -270,7 +270,7 @@ axom::Array<DoF, 2, axom::MemorySpace::Host> GetElementRestriction(const mfem::F
     return axom::Array<DoF, 2, axom::MemorySpace::Host>{};
   } else {
     uint64_t                                     dofs_per_elem = elem_dofs.size() / n;
-    axom::Array<DoF, 2, axom::MemorySpace::Host> output({n, dofs_per_elem});
+    axom::Array<DoF, 2, axom::MemorySpace::Host> output(n, dofs_per_elem);
```
Comment (jamiebramwell) [inline src/serac/numerics/functional/element_restriction.cpp]:
```suggestion
    axom::Array<DoF, 2, axom::MemorySpace::Host> output({n, dofs_per_elem});
```

File: src/serac/numerics/functional/element_restriction.cpp [hunk 2]
```diff
@@ -381,7 +381,7 @@ axom::Array<DoF, 2, axom::MemorySpace::Host> GetFaceDofs(const mfem::FiniteEleme
     return axom::Array<DoF, 2, axom::MemorySpace::Host>{};
   } else {
     uint64_t                                     dofs_per_face = face_dofs.size() / n;
-    axom::Array<DoF, 2, axom::MemorySpace::Host> output({n, dofs_per_face});
+    axom::Array<DoF, 2, axom::MemorySpace::Host> output(n, dofs_per_face);
```
Comment (jamiebramwell) [inline src/serac/numerics/functional/element_restriction.cpp]:
```suggestion
    axom::Array<DoF, 2, axom::MemorySpace::Host> output({n, dofs_per_face});
```

**Review by samuelpmishLLNL — APPROVED** [review]

File: src/serac/numerics/functional/element_restriction.cpp (hunk 2 above)
Comment (samuelpmishLLNL) [reply src/serac/numerics/functional/element_restriction.cpp]: out of curiosity, why does this change keep showing up in every pr and getting reverted?

Comment (jamiebramwell) [reply src/serac/numerics/functional/element_restriction.cpp]: I updated axom in my local environment at some point (but haven't pushed that changed through a PR). The initializer list constructor for `axom::Array` no longer exists in develop.

**Review by white238 — APPROVED** [review]

**Pull request merged**